File: hibernate/src/main/java/com/example/hibernate/model/Autobus.java
package com.example.hibernate.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "autobuses")
public class Autobus {

    @Id
    private String matricula;
    private Integer a_fabricacion;

    @OneToMany(mappedBy = "matricula", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Set<Visita> visistas;
}

File: angular/src/assets/Classes/Autobus.ts
import { Visita } from "./Visita";

export class AutoBus{
   public matricula:String;

   public a_fabricacion:Date;

   public visitas: Visita[];

   constructor(m:String, d:Date){
    this.matricula = m;
    this.a_fabricacion=d;
    this.visitas = [];
   }
}